src/backend/expungeservice/expunger/analyzers/time_analyzer.py

```diff
@@ -133,6 +135,30 @@ def _mark_as_time_ineligible(charges: List[Charge], reason: str, eligibility_dat
         for charge in charges:
             charge.set_time_ineligible(reason, eligibility_date)
 
+    @staticmethod
+    def _set_arrests_as_eligible(charges_with_summary: ChargesWithSummary):
+        # If most recent conviction is time eligible
+        # AND all other charges in the same case are arrests
+        # Then make all other charges in the case time eligible
+        mrc = charges_with_summary.most_recent_conviction
+        if not mrc is None and not mrc.expungement_result.time_eligibility is None and mrc.expungement_result.time_eligibility.status is EligibilityStatus.ELIGIBLE:
+            if charges_with_summary.second_most_recent_conviction:
+                pass
+            else:
+                for charge in mrc.case()().charges:
+                    if charge.acquitted():
+                        charge.set_time_eligible()
+        else:
+            if charges_with_summary.second_most_recent_conviction and not mrc is None:
+                for charge in mrc.case()().charges:
+                    if not charges_with_summary.second_most_recent_conviction.disposition is None:
+                        if charge.acquitted():
```

Comment:
I think you can combine this to

`if second_mrc.disposition and charge.acquitted()`